src/modules/appointments/appointments.controller.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Delete,
+  Body,
+  Param,
+  UseGuards,
+} from '@nestjs/common';
+import { CreateAppointmentDto } from '@modules/appointments/dto/create-appointment.dto';
+import { Appointment } from '@entities/Appointments';
+import { ApiTags, ApiOperation, ApiResponse } from '@nestjs/swagger';
+import { AuthGuard } from '@nestjs/passport';
+import { AppointmentsService } from './appointments.service';
+
+@ApiTags('appointments')
+@Controller('appointments')
+@UseGuards(AuthGuard('jwt'))
+export class AppointmentsController {
+  constructor(private readonly appointmentsService: AppointmentsService) { }
+
+  @Get()
+  @ApiOperation({ summary: 'Get all appointments' })
+  @ApiResponse({
+    status: 200,
+    description: 'Returns an array of appointments',
+    type: Appointment,
+    isArray: true,
+  })
+  async getAllAppointments(): Promise<Appointment[]> {
```

Comment:
If this is a route for step1 booking appointment, then it should be restricted to local doctor.
same with create route (check other routes too)

Reply:
add guards